### ISSUE TITLE
toml provider file discovery issue?

### ISSUE THREAD

**Issue by dtanner** [uchuhimo/konf#50]
Hi - first time using this (it's really nice BTW), and I think I found a minor issue. 

using the auto-extension discovery feature, it looks like it's limited to `json` and `properties` file types currently. e.g. If I send a `.toml` file to this function:
```kotlin
fun loadConfig(configFile: File): AppConfig {
    return Config()
        .from.file(configFile)
        .toValue()
}
```
it fails with `com.uchuhimo.konf.source.UnsupportedExtensionException: cannot detect supported extension for "test-config.toml", supported extensions: conf, json, properties, toml, xml, yml, yaml`.  
It looks like this is because of DefaultLoaders:137, where it calls 
```kotlin
 fun of(extension: String): Provider? =
            extensionToProvider[extension]
```
whose implementation is this:
```kotlin
private val extensionToProvider = ConcurrentHashMap(mutableMapOf(
            "json" to JsonProvider,
            "properties" to PropertiesProvider
        ))
```

my sample loadConfig works fine if I use `.from.toml.file(configFile)`.
See anything I'm doing wrong?  It's not a big deal either way.  Thanks.

**Comment by anelam** [comment]
I have the same problem, using version 0.22.1.

**Comment by uchuhimo** [comment]
@dtanner @anelam I have released a new version v0.23.0 that fixes this bug. You can try it now. Any further feedback is welcome.
